## Importing required librarirs for the analysis

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#### Importing data into read CSV file

In [18]:
data = pd.read_csv("C:/Users/sinch/Downloads/European_Bank.csv", encoding="ISO-8859-1")

#### Understanding our data 

In [19]:
data.head(10)

,Year,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
0,2025,15634602,Hargrave,619,France,Female,42,2,0.00,1,1,1,101348.88,1
1,2025,15647311,Hill,608,Spain,Female,41,1,83807.86,1,0,1,112542.58,0
2,2025,15619304,Onio,502,France,Female,42,8,159660.80,3,1,0,113931.57,1
3,2025,15701354,Boni,699,France,Female,39,1,0.00,2,0,0,93826.63,0
4,2025,15737888,Mitchell,850,Spain,Female,43,2,125510.82,1,1,1,79084.10,0
5,2025,15574012,Chu,645,Spain,Male,44,8,113755.78,2,1,0,149756.71,1
6,2025,15592531,Bartlett,822,France,Male,50,7,0.00,2,1,1,10062.80,0
7,2025,15656148,Obinna,376,Germany,Female,29,4,115046.74,4,1,0,119346.88,1
8,2025,15792365,He,501,France,Male,44,4,142051.07,2,0,1,74940.50,0
9,2025,15592389,H?,684,France,Male,27,2,134603.88,1,1,1,71725.73,0


In [20]:
data.shape

(10000, 14)

###### From the data we have observed that they have 10000 columns and 14 rows.
###### Which represents the 10000 customers of European bank and their details which is present in rows like CustomerId, Surname, CreditScore, Geography, Gender Age Tenure, Balance, NumOfProducts, Har


#### Finding missing value in the data

In [30]:
data.isnull().sum()

Year               0
CustomerId         0
Surname            0
CreditScore        0
Geography          0
Gender             0
Age                0
Tenure             0
Balance            0
NumOfProducts      0
HasCrCard          0
IsActiveMember     0
EstimatedSalary    0
Exited             0
dtype: int64

In [ ]:
###### No rows has miss

In [21]:
(data['Age'] == 0).sum()

np.int64(0)

In [25]:
(data['Year'] == 0).sum()

np.int64(0)

In [26]:
(data['CustomerId'] == 0).sum()

np.int64(0)

In [27]:
(data['CreditScore'] == 0).sum()

np.int64(0)

In [28]:
(data['Tenure'] == 0).sum()

np.int64(413)

###### Tenure has 413 rows which has value 0 which means it has 413 customer who has joined the bank recently like less than a year

In [29]:
(data['NumOfProducts'] == 0).sum()

np.int64(0)

###### This shows that each customer has atleast 1 product with the bank